app/utilities/PersistenceManager.java

```diff
@@ -54,6 +54,28 @@ public User getUserFromSession(UserSession userSession){
         return null;
     }
 
+    public static void editRestaurant(int restaurantId, Restaurant tempRestaurant) {
+        Restaurant restaurant = Restaurant.find.where().eq("restaurantId", restaurantId).findUnique();
+        restaurant.setDeals(tempRestaurant.getDeals());
+        restaurant.setImage(tempRestaurant.getImage());
+        restaurant.setName(tempRestaurant.getName());
+        restaurant.setPhone(tempRestaurant.getPhone());
+        restaurant.setRating(tempRestaurant.getRating());
+        restaurant.setReservationPrice(tempRestaurant.getReservationPrice());
+        restaurant.setWorkingHours(tempRestaurant.getWorkingHours());
+
+        restaurant.getAddress().setStreetName(tempRestaurant.getAddress().getStreetName());
+        restaurant.getAddress().setCity(tempRestaurant.getAddress().getCity());
+        restaurant.getAddress().setCountry(tempRestaurant.getAddress().getCountry());
```

Comment:
I created those methods. But I don't see why updateData method should be generic, is there some benefit if that approach is used in this case, since it is possible to solve the issue with "public void updateData(Restaurant restaurant)" method declaration